File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converters/ConverterTime.java
package pl.szczodrzynski.edziennik.data.db.converters;

import androidx.room.TypeConverter;

import pl.szczodrzynski.edziennik.utils.models.Time;

public class ConverterTime {

    @TypeConverter
    public static Time toTime(String value) {
        return value == null ? null : value.equals("null") ? null : Time.fromHms(value);
    }

    @TypeConverter
    public static String toString(Time value) {
        return value == null ? null : value.getStringValue();
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterTime.kt
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import pl.szczodrzynski.edziennik.utils.models.Time

class ConverterTime {
    @TypeConverter
    fun toTime(value: String?): Time? = when (value) {
        null -> null
        "null" -> null
        else -> Time.fromHms(value)
    }

    @TypeConverter
    fun toString(value: Time?): String? = value?.stringValue
}
